components/script/dom/webgl/webglprogram.rs

```diff
@@ -24,15 +25,160 @@ use crate::dom::webgl::webglobject::WebGLObject;
 use crate::dom::webgl::webglrenderingcontext::{Operation, WebGLRenderingContext};
 use crate::dom::webgl::webglshader::WebGLShader;
 use crate::dom::webgl::webgluniformlocation::WebGLUniformLocation;
+use crate::dom::webglrenderingcontext::capture_webgl_backtrace;
 use crate::script_runtime::CanGc;
 
-#[dom_struct]
-pub(crate) struct WebGLProgram {
-    webgl_object: WebGLObject,
+#[derive(JSTraceable, MallocSizeOf)]
+struct DroppableWebGLProgram {
     #[no_trace]
     id: WebGLProgramId,
-    is_in_use: Cell<bool>,
+    context: WeakRef<WebGLRenderingContext>,
+    fragment_shader: Option<WeakRef<WebGLShader>>,
+    vertex_shader: Option<WeakRef<WebGLShader>>,
     marked_for_deletion: Cell<bool>,
+    is_in_use: Cell<bool>,
```

Comment:
I'm still skeptical that we need the marked_for_deletion and is_in_use fields inside DroppableWebGLProgram.